File: 695-max-area-of-island/695-max-area-of-island.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        row, col, ans = len(grid), len(grid[0]), -1
        def area(i, j):
            if i<0 or j<0 or i>=row or j>=col or grid[i][j]==0:
                return 0;
            grid[i][j]=0
            res= 1+area(i+1, j)+area(i-1, j)+area(i, j+1)+area(i, j-1)
            return res
        
        for x in range(row):
            for y in range(col):
                ans=max(ans, area(x, y))
        return ans
        

File: 695-max-area-of-island/695-max-area-of-island.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int ans=0;
        for(int i=0, m=grid.length; i<m; i++){
            for(int j=0, n=grid[0].length; j<n; j++){
                if(grid[i][j]==1) ans=Math.max(ans, area(grid, i, j));
            }
        }
        return ans;
    }
    int area(int[][] grid, int i, int j){
        if(i<0 || j<0 || i>=grid.length || j>=grid[0].length || grid[i][j]==0) return 0;
        grid[i][j]=0;
        return 1+area(grid, i+1, j)+area(grid, i-1, j)+area(grid, i, j+1)+area(grid, i, j-1);
    }
}